src/Microsoft.Extensions.RuntimeEnvironment.Sources/Native/PlatformApis.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.Internal.Native
+{
+    internal static class PlatformApis
+    {
+        private class DistroInfo
+        {
+            public string Id;
+            public string VersionId;
+        }
+
+        private static readonly Lazy<DistroInfo> _distroInfo = new Lazy<DistroInfo>(LoadDistroInfo);
+
+        public static string GetOSVersion()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return NativeMethods.Windows.RtlGetVersion() ?? string.Empty;
+            }
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```

Comment:
If we don't actually care about the version in non-Windows scenarios, could we just throw `PlatformNotSupported` here?